vpn/tunnel.go

```diff
@@ -42,6 +51,7 @@ func NewTunnel(
 		logger:          logger,
 		requestLoopDone: make(chan struct{}),
 	}
+	logger.AppendSinks(t)
```

Comment:
Loggers are passed by value, not reference, so this has no effect.  But, you have to be really careful about logging to the VPN protocol from any of this protocol machinery, since that could deadlock. I think we should just avoid this rather than try to work out when it is and isn't safe. Even if we got it right it would be a landmine for future contributors.

We still want the logger for the protocol machinery during unit tests, using `testutil.Logger(t)`.

In production, the logger can be empty of sinks.  At some future date if we find we want it, we could log to a ring buffer or something.